src/js/newTab.js

```diff
@@ -1,27 +1,61 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
-/*globals Request, CacheTasks, gGrid, gPage, gUpdater, gUserDatabase,
-  gPinnedLinks, gBlockedLinks, async*/
+/*globals Request, CacheTasks, gGrid, gPage, gUpdater, async*/
 
 "use strict";
 
 (function(exports) {
   const gNewTab = {
 
-    listeners: {},
+    listeners: new Map(),
 
     _l10nStrings: new Map(),
 
+    privateBrowsingMode: false,
```

Comment:
I made sure these properties were no longer added dynamically. Adding props dynamically to objects can be very confusing.